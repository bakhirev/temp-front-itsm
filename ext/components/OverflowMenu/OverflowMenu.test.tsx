import React from 'react';
import { shallow } from 'enzyme';

import { OverflowMenu } from './';

jest.mock('../common', () => ({
  DEFAULT_THEME: {},
  Z_INDEXES: {
    TOOLTIP: 777,
  },
}));

describe('OverflowMenu', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  const overflowMenuProps = {
    list: [
      { value: 1, label: 'Москва' },
      { value: 2, label: 'Новгород' },
      { value: 3, label: 'Орел' },
    ],
  };

  it('should render component with required properties', () => {
    const wrapper = shallow(<OverflowMenu {...overflowMenuProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with medium size', () => {
    const wrapper = shallow(<OverflowMenu {...overflowMenuProps} size={'medium'} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with small size', () => {
    const wrapper = shallow(<OverflowMenu {...overflowMenuProps} size={'small'} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with disabled state', () => {
    const wrapper = shallow(<OverflowMenu {...overflowMenuProps} disabled />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with right menu alignment', () => {
    const wrapper = shallow(<OverflowMenu {...overflowMenuProps} menu={{ alignment: 'right' }} />);
    expect(wrapper).toMatchSnapshot();
  });
});
