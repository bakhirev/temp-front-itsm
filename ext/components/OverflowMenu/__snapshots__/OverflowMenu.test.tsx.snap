// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverflowMenu should render component with disabled state 1`] = `
<StateManager
  components={
    Object {
      "ClearIndicator": null,
      "Control": [Function],
      "DropdownIndicator": [Function],
      "IndicatorsContainer": [Function],
      "Menu": [Function],
      "MenuList": [Function],
      "Option": [Function],
      "Placeholder": [Function],
      "SelectContainer": [Function],
      "SingleValue": [Function],
      "ValueContainer": [Function],
    }
  }
  defaultInputValue=""
  defaultMenuIsOpen={false}
  defaultValue={null}
  isDisabled={true}
  isFocused={false}
  isSearchable={false}
  menuAlignment="left"
  menuMarginTop={8}
  menuWidth={290}
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  options={
    Array [
      Object {
        "label": "Москва",
        "value": 1,
      },
      Object {
        "label": "Новгород",
        "value": 2,
      },
      Object {
        "label": "Орел",
        "value": 3,
      },
    ]
  }
  size={24}
  styles={
    Object {
      "dropdownIndicator": [Function],
      "indicatorSeparator": [Function],
    }
  }
/>
`;

exports[`OverflowMenu should render component with medium size 1`] = `
<StateManager
  components={
    Object {
      "ClearIndicator": null,
      "Control": [Function],
      "DropdownIndicator": [Function],
      "IndicatorsContainer": [Function],
      "Menu": [Function],
      "MenuList": [Function],
      "Option": [Function],
      "Placeholder": [Function],
      "SelectContainer": [Function],
      "SingleValue": [Function],
      "ValueContainer": [Function],
    }
  }
  defaultInputValue=""
  defaultMenuIsOpen={false}
  defaultValue={null}
  isDisabled={false}
  isFocused={false}
  isSearchable={false}
  menuAlignment="left"
  menuMarginTop={8}
  menuWidth={290}
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  options={
    Array [
      Object {
        "label": "Москва",
        "value": 1,
      },
      Object {
        "label": "Новгород",
        "value": 2,
      },
      Object {
        "label": "Орел",
        "value": 3,
      },
    ]
  }
  size={20}
  styles={
    Object {
      "dropdownIndicator": [Function],
      "indicatorSeparator": [Function],
    }
  }
/>
`;

exports[`OverflowMenu should render component with required properties 1`] = `
<StateManager
  components={
    Object {
      "ClearIndicator": null,
      "Control": [Function],
      "DropdownIndicator": [Function],
      "IndicatorsContainer": [Function],
      "Menu": [Function],
      "MenuList": [Function],
      "Option": [Function],
      "Placeholder": [Function],
      "SelectContainer": [Function],
      "SingleValue": [Function],
      "ValueContainer": [Function],
    }
  }
  defaultInputValue=""
  defaultMenuIsOpen={false}
  defaultValue={null}
  isDisabled={false}
  isFocused={false}
  isSearchable={false}
  menuAlignment="left"
  menuMarginTop={8}
  menuWidth={290}
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  options={
    Array [
      Object {
        "label": "Москва",
        "value": 1,
      },
      Object {
        "label": "Новгород",
        "value": 2,
      },
      Object {
        "label": "Орел",
        "value": 3,
      },
    ]
  }
  size={24}
  styles={
    Object {
      "dropdownIndicator": [Function],
      "indicatorSeparator": [Function],
    }
  }
/>
`;

exports[`OverflowMenu should render component with right menu alignment 1`] = `
<StateManager
  components={
    Object {
      "ClearIndicator": null,
      "Control": [Function],
      "DropdownIndicator": [Function],
      "IndicatorsContainer": [Function],
      "Menu": [Function],
      "MenuList": [Function],
      "Option": [Function],
      "Placeholder": [Function],
      "SelectContainer": [Function],
      "SingleValue": [Function],
      "ValueContainer": [Function],
    }
  }
  defaultInputValue=""
  defaultMenuIsOpen={false}
  defaultValue={null}
  isDisabled={false}
  isFocused={false}
  isSearchable={false}
  menuAlignment="right"
  menuMarginTop={8}
  menuWidth={290}
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  options={
    Array [
      Object {
        "label": "Москва",
        "value": 1,
      },
      Object {
        "label": "Новгород",
        "value": 2,
      },
      Object {
        "label": "Орел",
        "value": 3,
      },
    ]
  }
  size={24}
  styles={
    Object {
      "dropdownIndicator": [Function],
      "indicatorSeparator": [Function],
    }
  }
/>
`;

exports[`OverflowMenu should render component with small size 1`] = `
<StateManager
  components={
    Object {
      "ClearIndicator": null,
      "Control": [Function],
      "DropdownIndicator": [Function],
      "IndicatorsContainer": [Function],
      "Menu": [Function],
      "MenuList": [Function],
      "Option": [Function],
      "Placeholder": [Function],
      "SelectContainer": [Function],
      "SingleValue": [Function],
      "ValueContainer": [Function],
    }
  }
  defaultInputValue=""
  defaultMenuIsOpen={false}
  defaultValue={null}
  isDisabled={false}
  isFocused={false}
  isSearchable={false}
  menuAlignment="left"
  menuMarginTop={8}
  menuWidth={290}
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  options={
    Array [
      Object {
        "label": "Москва",
        "value": 1,
      },
      Object {
        "label": "Новгород",
        "value": 2,
      },
      Object {
        "label": "Орел",
        "value": 3,
      },
    ]
  }
  size={16}
  styles={
    Object {
      "dropdownIndicator": [Function],
      "indicatorSeparator": [Function],
    }
  }
/>
`;
